documentation/source/newsfragments/README.rst

```diff
@@ -0,0 +1,46 @@
+:orphan:
+
+Release Notes
+=============
+
+We are using `towncrier <https://pypi.org/project/towncrier/>`_ to handle
+our relase notes, and this directory  contains the input for it -
+"news fragments" which are short files that contain a small
+**ReST**-formatted text that will be added to the next version's
+Release Notes page.
+
+Make sure to use full sentences with correct case and punctuation,
+and use Sphinx references
+(see https://sphinx-tutorial.readthedocs.io/cheatsheet/).
+
+Each file should be named like ``<PULL REQUEST>.<TYPE>.rst``, where
+``<PULL REQUEST>`` is a pull request number, and ``<TYPE>`` is one of:
+
+* ``feature``: New user-facing feature.
+* ``bugfix``: A bug fix.
+* ``doc``: Documentation improvement.
+* ``removal``: Deprecation or removal of public API.
+* ``misc``: A ticket has been closed, but it is not of interest to users.
+
+.. comment_numpy_has_these:
+    * ``new_function``: New user facing functions.
+    * ``deprecation``: Changes existing code to emit a DeprecationWarning.
+    * ``future``: Changes existing code to emit a FutureWarning.
+    * ``expired``: Removal of a deprecated part of the API.
+    * ``compatibility``: A change which requires users to change code and is not
+      backwards compatible. (Not to be used for removal of deprecated features.)
+    * ``c_api``: Changes in the Numpy C-API exported functions
+    * ``new_feature``: New user facing features like ``kwargs``.
+    * ``improvement``: Performance and edge-case changes
+    * ``change``: Other changes
+    * ``highlight``: Adds a highlight bullet point to use as a possibly highlight
+      of the release.
+
+You can install ``towncrier`` and run ``towncrier --draft``
+if you want to get a preview of how your change will look in the final release
+notes.
```

Comment:
I think we need to add a brief section somewhere as instructions for maintainers about how to assemble a release.

Reply:
Once this PR is accepted I will add that information to https://github.com/cocotb/cocotb/wiki/Release-Process

Reply:
Why doesn't towncrier always run when creating the docs? This way we can also see if the fragments were added correctly by previewing the RTD page that's generated for each PR.

Reply:
I'm adding that.